### PR TITLE
Added guard code for potentially

### PR DESCRIPTION
**Pull request by chrisjowen**
Referer URL missing if the user directly hits a page that requires registration directly from the url bar. Guard code added, also crazy nested option removed.

### CODE REVIEW

File: frontend/app/actions/RegistrationUri.scala [hunk 1]
```diff
@@ -9,43 +9,38 @@ object RegistrationUri {
...
-      case `supporterTier` => Some("SUP")
-      case `partnerTier` => Some("PAR")
-      case `patronTier` => Some("PAT")
-      case `friendTier` => Some("FRI")
-      case _ => None
-    }
-    campaignTier.map {
-      tier => Some(Seq(CAMPAIGN_SOURCE, campaignReferer, tier).mkString("_"))
-    }
+    referingPageCodes.get(refererUrl.path)
```
Comment (ostapneko) [inline frontend/app/actions/RegistrationUri.scala]: While we are at it, I would guard the `Uri.parse` as well with like `Try(Uri.parse).toOption`

File: frontend/app/actions/RegistrationUri.scala [hunk 2]
```diff
@@ -9,43 +9,38 @@ object RegistrationUri {
...
+    redirectUrl.concat(s"&INTCMP=$campaignCode")
   }
 
-  private def extractCampaignCode(referer: String, path: String) = {
+  private def extractCampaignCode(refererOpt: Option[String], path: String) : String = {
+
+    val referingCode = refererOpt.flatMap(getReferingPageCode)
+    val campaignTier: Option[String] = campaignTierCodes.get(path)
+
+    (Seq(CAMPAIGN_SOURCE) ++ referingCode ++ campaignTier).mkString("_")
```
Comment (ostapneko) [inline frontend/app/actions/RegistrationUri.scala]: Does it rely on implicit conversion from `Option` to `Seq`?

If so, then we can easily have an inconsistent code without a referingCode.

What about a simple `Option`-based for-comprehension instead?

Comment (chrisjowen) [inline frontend/app/actions/RegistrationUri.scala]: Regarding the fail case, yes thats exactly what we want to happen at least they want to know the source is Membership. Nope no Option to Seq implicit conversion going on here, internal magic of IndexedSeqLike pimipery. What advantage would I get form for comprehension here?

Comment (ostapneko) [inline frontend/app/actions/RegistrationUri.scala]: Well if we are happy to send an inconsistent code, clearly there is no need for for-comprehension

**Comment by ostapneko** [comment]
:+1:

**Comment by chrisjowen** [comment]
Tested on production, redirects witout error without referer

**Pull request merged**